vpr/src/backend/physical_netlist_writer.h

```diff
@@ -0,0 +1,246 @@
+#ifndef WRITE_PHYSICAL_NETLIST_H
+#define WRITE_PHYSICAL_NETLIST_H
+
+#include <iostream>
+#include <list>
+#include <map>
+#include <ostream>
+#include <set>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "DeviceResources.capnp.h"
+#include "PhysicalNetlist.capnp.h"
+#include "capnp/serialize.h"
+#include "capnp/serialize-packed.h"
+#include "fpga_interchange_arch_utils.h"
+#include "netlist_walker.h"
+#include "netlist_writer.h"
+#include "route_tree_type.h"
+
+using namespace DeviceResources;
+using namespace capnp;
+
+/** @brief Writes Physical netlist interchange file based on the netlist name by walking the netlist.
+ */
+void write_physical_netlist(t_vpr_setup& vpr_setup);
+
+namespace ic {
+
+struct StringEnumerator {
+    std::vector<std::string> strings;
+    std::unordered_map<std::string, size_t> string_to_index;
+
+    size_t get_index(const std::string& s) {
+        auto result = string_to_index.emplace(s, strings.size());
+        if (result.second) {
+            // This string was inserted, append.
+            strings.push_back(s);
+        }
+
+        return result.first->second;
+    }
+
+    std::string get_string(size_t idx) {
+        return strings[idx];
+    }
+};
+
+enum e_route_type {
```

Comment:
```suggestion
enum class e_route_type {
```

Reply:
Using `class` actually incurs into compile errors, I left it as is